### PR TITLE
Add optional cache TTL to fenceposting

### PR DESCRIPTION
**Pull request by kchro3**
### What and why? 🤔

The TTL for the fenceposting cache should be configurable by the caller.

### Testing Steps ✍️

Code review should be fine, since this is a simple pass-through. But we'll see how much coverage changes.

### You're it! 👑

@Automattic/stream-builders

### CODE REVIEW

File: lib/Tumblr/StreamBuilder/FencepostRanking/CacheFencepostProvider.php [hunk 1]
```diff
@@ -37,13 +37,19 @@ final class CacheFencepostProvider extends FencepostProvider
      */
     private $cache;
 
+    /**
+     * @var int TTL in seconds. Defaults to 1 week.
```
Comment (cyle) [inline lib/Tumblr/StreamBuilder/FencepostRanking/CacheFencepostProvider.php]:
```suggestion
     * @var int Default cached value TTL in seconds.
```

File: lib/Tumblr/StreamBuilder/FencepostRanking/CacheFencepostProvider.php [hunk 2]
```diff
@@ -37,13 +37,19 @@ final class CacheFencepostProvider extends FencepostProvider
...
+     */
+    private $ttl_seconds;
+
     /**
      * CacheFencepostProvider constructor.
      * @param CacheProvider $cache Provider of cache
```
Comment (cyle) [inline lib/Tumblr/StreamBuilder/FencepostRanking/CacheFencepostProvider.php:47]:
```suggestion
     * @param CacheProvider $cache Provider of cache
     * @param int $ttl_seconds Default cached value TTL in seconds.
```

File: lib/Tumblr/StreamBuilder/FencepostRanking/CacheFencepostProvider.php [hunk 3]
```diff
@@ -37,13 +37,19 @@ final class CacheFencepostProvider extends FencepostProvider
      */
     private $cache;
 
+    /**
+     * @var int TTL in seconds. Defaults to 1 week.
+     */
+    private $ttl_seconds;
```
Comment (cyle) [inline lib/Tumblr/StreamBuilder/FencepostRanking/CacheFencepostProvider.php]:
```suggestion
    private int $ttl_seconds;
```

i think we can, and should, use typehints when possible for new code

File: lib/Tumblr/StreamBuilder/FencepostRanking/CacheFencepostProvider.php [hunk 4]
```diff
@@ -95,7 +101,8 @@ public function set_latest_timestamp(string $fence_id, int $timestamp_ms)
         $this->cache->set(
             CacheProvider::OBJECT_TYPE_FENCEPOST_ID,
             self::cachekey_fence($fence_id),
-            $timestamp_ms
+            $timestamp_ms,
+            $this->ttl_seconds
```
Comment (cyle) [inline lib/Tumblr/StreamBuilder/FencepostRanking/CacheFencepostProvider.php:106]: i think `set_fencepost_epoch()` needs to be updated as well

Comment (lucila) [reply lib/Tumblr/StreamBuilder/FencepostRanking/CacheFencepostProvider.php:106]: also `set_fencepost`

**Review by lucila — APPROVED** [review]

**Review by cyle — APPROVED** [review]

**Pull request merged**